placement/algo/sharded_helper.go

```diff
@@ -414,60 +447,17 @@ func (ph *placementHelper) returnInitializingShardsToSource(
 		}
 		sourceInstance, ok := candidateMap[sourceID]
 		if !ok {
-			return fmt.Errorf("could not find sourceID %s for instance %s, shard %d", sourceID, from.ID(), s.ID())
+			// NB(cw): This is not an error because the candidates are not
```

Comment:
When this happens, these shards won't be able to be returned to the source instance, is that ok?

Reply:
That's ok, if you pass in all the instances in the placement then this won't happen, for cases when you only want to move shards to a subset of instances by calling PlaceShards(), shards will be moved to those instances even though they are not the source instance for the Initializing shards.

Reply:
Sounds good.